cog/sql_abstract.py

```
@@ -0,0 +1,170 @@
+DataclassT = TypeVar("DataclassT")
+
+
+# TODO: optimize with attr module
+class AttributeKeyedDict(UserDict, Generic[DataclassT]):
+    def __init__(self, primary_key_name: str, *args, **kwargs):
+        if not is_dataclass(DataclassT):
+            raise TypeError(
+                "KeyedDict should be specified a dataclass as the type of the item."
+            )
+
+        if not primary_key_name in DataclassT.__annotations__:
```

Ruff (0.8.2)

Test for membership should be `not in`

Convert to `not in`

(E713)